addons/l10n_sa_edi/models/account_move.py

```diff
@@ -195,35 +197,37 @@ def _l10n_sa_log_results(self, xml_content, response_data=None, error=False):
                 'mimetype': 'application/xml',
             })
             bootstrap_cls, title = ("danger", _("Invoice was rejected by ZATCA"))
-            error_msg = response_data['error']
-            content = Markup("""
-                <p class='mb-0'>
-                    %s
-                </p>
-                <hr>
-                <p class='mb-0'>
-                    %s
-                </p>
-            """) % (_('The invoice was rejected by ZATCA. Please, check the response below:'), error_msg)
+            subtitle = _('The invoice was rejected by ZATCA. Please, check the response below:')
+            content = response_data['error']
         if response_data and response_data.get('validationResults', {}).get('warningMessages'):
-            status_code = response_data.get('status_code')
             bootstrap_cls, title = ("warning", _("Invoice was Accepted by ZATCA (with Warnings)"))
-            content = Markup("""
-                <p class='mb-0'>
-                    %s
-                </p>
-                <hr>
-                <p class='mb-0'>
-                    <b>%s</b>%s
-                </p>
-            """) % (_('The invoice was accepted by ZATCA, but returned warnings. Please, check the response below:'),
-                    f"[{status_code}] " if status_code else "",
-                    Markup("<br/>").join([Markup("<b>%s</b> : %s") % (m['code'], m['message']) for m in response_data['validationResults']['warningMessages']]))
+            subtitle = _('The invoice was accepted by ZATCA, but returned warnings. Please, check the response below:')
+            content = Markup("""<b>%s</b>%s""") % (f"[{status_code}] " if status_code else "", 
+                    Markup("<br/>").join([Markup("<b>%s</b> : %s") % (m['code'], m['message']) for m in response_data['validationResults']['warningMessages']])
+            )
+        if response_data.get("error") and response_data.get("excepted"):
+            bootstrap_cls, title = ("warning", _("Warning: Unable to Retrieve a Response from ZATCA"))
+            subtitle = _('Unable to retrieve response from ZATCA. Please, check the response below:')
+            content = content = response_data['error']
+        if status_code == 409:                  
+            bootstrap_cls, title = ("warning", _("Warning: Invoice was already successfully reported to ZATCA"))
+            subtitle = _("This invoice was already successfully reported to ZATCA. Please, check the response below:")
+            content = Markup("""<b>%s</b>%s""") % (f"[{status_code}] " if status_code else "", 
+                Markup("<br/>").join([Markup("<b>%s</b> : %s") % (m['code'], m['message']) for m in response_data['validationResults']['errorMessages']])
+            )
```

Comment:
indent is a bit awful there ahah what about doing something like this ?
<img width="840" height="182" alt="image" src="https://github.com/user-attachments/assets/0e10fbc1-afaf-4647-86fa-87906c433949" />